roles/get-requirements.yaml

```
@@ -0,0 +1,154 @@
+- name: Download compliantkubernetes-apps requirements
+  hosts: localhost
+  vars:
+    install_path: /usr/local/bin
+    install_user: "{{ lookup('env','USER') }}"
+    kubectl_version: 1.24.4
+    helm_version: 3.8.0
+    helmfile_version: 0.146.0
+    helmdiff_version: 3.5.0
+    helmsecrets_version: 3.12.0
+    jq_version: 1.6
+    s3cmd_version: 2.*
+    sops_version: 3.7.3
+    yq_version: 3.4.1
+    yq4_version: 4.26.1
+    pwgen_version: 2.08*
+    apache2_utils_version: 2.4.*
+    apt_list:
+      - curl
+      - s3cmd={{ s3cmd_version }}
+      - dnsutils
+      - apache2-utils={{ apache2_utils_version }}
+      - pwgen={{ pwgen_version }}
+    get_url_list:
+      - command: jq
+        version: "{{ jq_version }}"
+        url: https://github.com/stedolan/jq/releases/download/jq-{{ jq_version }}/jq-linux64
+        dest: "{{ install_path }}/jq"
+      - command: sops
+        version: "{{ sops_version }}"
+        url: https://github.com/mozilla/sops/releases/download/v{{ sops_version }}/sops-v{{ sops_version }}.linux
+        dest: "{{ install_path }}/sops"
+      - command: yq
+        version: "{{ yq_version }}"
+        url: https://github.com/mikefarah/yq/releases/download/{{ yq_version }}/yq_linux_amd64
+        dest: "{{ install_path }}/yq"
+      - command: yq4
+        version: "{{ yq4_version }}"
+        url: https://github.com/mikefarah/yq/releases/download/v{{ yq4_version }}/yq_linux_amd64
+        dest: "{{ install_path }}/yq4"
+
+  connection: local
+  become: yes
+  become_user: root
+  tasks:
+    - name: Install apt-packages
+      apt:
+        pkg: "{{ apt_list }}"
+
+    - name: Kubectl
+      block:
+        - name: Check if Kubectl exists
+          stat:
+            path: "{{ install_path }}/kubectl"
+          register: kubectl_exists
+
+        - name: Check Kubectl
+          command: "kubectl version -ojson"
```

```suggestion
          command: "kubectl version --client -ojson"
```
does it make sense to check on client?

---

Yes, without "--client" I get an error. Although I do also get the correct output, so it might work but it doesn't look great that we get an error.
```
TASK [Check Kubectl]
fatal: [127.0.0.1]: FAILED! => changed=true                                                                                                                              
  cmd:                                                                                                                                                                   
  - kubectl                                                                                                                                                              
  - version                                                                                                                                                              
  - -ojson
...
stderr: The connection to the server localhost:8080 was refused - did you specify the right host or port?
  stdout: |-                                                                                                                                                             
    {                                                                                                                                                                    
      "clientVersion": {                                                                                                                                                 
        "major": "1",                                                                                                                                                    
        "minor": "24",                                                                                                                                                   
        "gitVersion": "v1.24.4",                                                                                                                                         
        "gitCommit": "95ee5ab382d64cfe6c28967f36b53970b8374491",                                                                                                         
        "gitTreeState": "clean",                                                                                                                                         
        "buildDate": "2022-08-17T18:54:23Z",                                                                                                                             
        "goVersion": "go1.18.5",                                                                                                                                         
        "compiler": "gc",                                                                                                                                                
        "platform": "linux/amd64"                                                                                                                                        
      },                                                                                                                                                                 
      "kustomizeVersion": "v4.5.4"                                                                                                                                       
    }
...
```

---

I was looking for that --client. But i missread it. I will add it.